### PR TITLE
[ingress upgrade] Poll for GCP resources after node upgrade

### LINKED ISSUE #58839: [flaky test] kube-proxy migration jobs often fail on ingress resources comparision
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network
/area test

**What happened**:
kube-proxy migration jobs are flaky since the ingress resource comparison logic was added:
- https://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-latest-upgrade-kube-proxy-ds
- https://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-latest-downgrade-kube-proxy-ds

An example error as below:
```
[sig-cluster-lifecycle] kube-proxy migration [Feature:KubeProxyDaemonSetMigration] Upgrade kube-proxy from static pods to a DaemonSet should maintain a functioning cluster [Feature:KubeProxyDaemonSetUpgrade] 21m53s

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/lifecycle/cluster_upgrade.go:272
Expected error:
    <*errors.errorString | 0xc4214e16c0>: {
        s: "resources after ingress upgrade were different:\n Pre-Upgrade: ([]*compute.InstanceGroup)[<*>{CreationTimestamp:(string)2018-01-25T12:59:49.164-08:00 Description:(string) Fingerprint:(string)6Y5A0R5ln3o= Id:(uint64)1103185785546865578 Kind:(string)compute#instanceGroup Name:(string)k8s-ig--211370289044f4f4 NamedPorts:([]*compute.NamedPort)[<*>{Name:(string)port30855 Port:(int64)30855 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>} <*>{Name:(string)port32020 Port:(int64)32020 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}] Network:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/global/networks/bootstrap-e2e Region:(string) SelfLink:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f/instanceGroups/k8s-ig--211370289044f4f4 Size:(int64)3 Subnetwork:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/regions/us-central1/subnetworks/bootstrap-e2e Zone:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f ServerResponse:(googleapi.ServerResponse){HTTPStatusCode:(int)0 Header:(http.Header)<nil>} ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}]\n Post-Upgrade: ([]*compute.InstanceGroup)[<*>{CreationTimestamp:(string)2018-01-25T12:59:49.164-08:00 Description:(string) Fingerprint:(string)6Y5A0R5ln3o= Id:(uint64)1103185785546865578 Kind:(string)compute#instanceGroup Name:(string)k8s-ig--211370289044f4f4 NamedPorts:([]*compute.NamedPort)[<*>{Name:(string)port30855 Port:(int64)30855 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>} <*>{Name:(string)port32020 Port:(int64)32020 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}] Network:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/global/networks/bootstrap-e2e Region:(string) SelfLink:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f/instanceGroups/k8s-ig--211370289044f4f4 Size:(int64)2 Subnetwork:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/regions/us-central1/subnetworks/bootstrap-e2e Zone:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f ServerResponse:(googleapi.ServerResponse){HTTPStatusCode:(int)0 Header:(http.Header)<nil>} ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}]",
    }
    resources after ingress upgrade were different:
     Pre-Upgrade: ([]*compute.InstanceGroup)[<*>{CreationTimestamp:(string)2018-01-25T12:59:49.164-08:00 Description:(string) Fingerprint:(string)6Y5A0R5ln3o= Id:(uint64)1103185785546865578 Kind:(string)compute#instanceGroup Name:(string)k8s-ig--211370289044f4f4 NamedPorts:([]*compute.NamedPort)[<*>{Name:(string)port30855 Port:(int64)30855 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>} <*>{Name:(string)port32020 Port:(int64)32020 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}] Network:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/global/networks/bootstrap-e2e Region:(string) SelfLink:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f/instanceGroups/k8s-ig--211370289044f4f4 Size:(int64)3 Subnetwork:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/regions/us-central1/subnetworks/bootstrap-e2e Zone:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f ServerResponse:(googleapi.ServerResponse){HTTPStatusCode:(int)0 Header:(http.Header)<nil>} ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}]
     Post-Upgrade: ([]*compute.InstanceGroup)[<*>{CreationTimestamp:(string)2018-01-25T12:59:49.164-08:00 Description:(string) Fingerprint:(string)6Y5A0R5ln3o= Id:(uint64)1103185785546865578 Kind:(string)compute#instanceGroup Name:(string)k8s-ig--211370289044f4f4 NamedPorts:([]*compute.NamedPort)[<*>{Name:(string)port30855 Port:(int64)30855 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>} <*>{Name:(string)port32020 Port:(int64)32020 ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}] Network:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/global/networks/bootstrap-e2e Region:(string) SelfLink:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f/instanceGroups/k8s-ig--211370289044f4f4 Size:(int64)2 Subnetwork:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/regions/us-central1/subnetworks/bootstrap-e2e Zone:(string)https://www.googleapis.com/compute/v1/projects/k8s-jkns-gce-ingress1-4/zones/us-central1-f ServerResponse:(googleapi.ServerResponse){HTTPStatusCode:(int)0 Header:(http.Header)<nil>} ForceSendFields:([]string)<nil> NullFields:([]string)<nil>}]
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/ingress.go:185
```

Turned out because during node upgrade, nodes might be removed and re-added to the instance group, which takes time, hence the first resources comparison may fail. Instead of doing an one-off comparison, we could do polling instead.

/assign

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
During node upgrade, nodes might be removed and re-added to the instance group, which takes time hence the first comparison may fail. This PR makes ingress upgrade test polls for GCP resources after node upgrade.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #58839 

**Special notes for your reviewer**:
/assign @rramkumar1 @bowei 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rramkumar1** [comment]
LGTM

File: test/e2e/upgrades/ingress.go [hunk 1]
```diff
@@ -141,7 +144,7 @@ func (t *IngressUpgradeTest) Teardown(f *framework.Framework) {
 	framework.CleanupGCEIngressController(t.gceController)
 }
 
-func (t *IngressUpgradeTest) verify(f *framework.Framework, done <-chan struct{}, testDuringDisruption bool) {
+func (t *IngressUpgradeTest) verify(f *framework.Framework, done <-chan struct{}, testDuringDisruption, pollForGCPResources bool) {
```
Comment (bowei) [inline test/e2e/upgrades/ingress.go]: can we have 
```
type ingressUpgradeVerifyOptions int

const (
  testDuringDisruption ingressUpgradeVerifyOptions = 1 << iota
  pollForGCPResources ingressUpgradeVerifyOptions = 1 << iota
)

func ... verify(f, done, ingressUpgradeVerifyOptions)
```

Comment (MrHohn) [reply test/e2e/upgrades/ingress.go]: Thanks for the suggestion, but these verify options might not fit into this switch option model. `testDuringDisruption` and `pollForGCPResources` are not necessarily mutually exclusive, just that we don't have a test case that requires them to happen at the same time yet.

**Comment by MrHohn** [comment]
Turned out https://github.com/kubernetes/kubernetes/pull/59077 already fixed the upgrade/downgrade kube-proxy jobs because we now properly wait for the ingress sync to happen first.

We may not need this anymore.

**Pull request closed without merging**